### PR TITLE
Added contribution instructions

### LINKED ISSUE #86: Document contribution
We need to add a CONTRIBUTING.md explaining how the community can contribute to Clay

### PR DESCRIPTION
**Pull request by keegan-lillo**
resolves #86

### CODE REVIEW

**Comment by keegan-lillo** [comment]
@cjwilliams @cat-haines Can you please review this and make sure this all looks okay?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+If there is currently no GitHub issue open for your proposed contribution,
+then please make one. This allows us to assign milestones and triage priority.
+
+### 3. Install the project dependencies.
+
+`$ npm install`
+
+### 4. Develop your feature
+
+Commit as often as you like, however please squash commits that fix typos or are
```
Comment (cat-haines) [inline CONTRIBUTING.md]: The language in this doc + the piece-of-cake tag seems to indicate that we're really trying to encourage new developers to contribute (which is awesome - thank you)..

Asking developers to squash commits might be a bit intimidating for new devs and discourage them from participating.. could we either remove this section (and talk about how to do it in PRs that need it), or add more information / a link to (a) what this means, and (b) how to do it?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+otherwise noisy.
+
+### 5. Run the tests
+
+`$ npm run test`
```
Comment (cat-haines) [inline CONTRIBUTING.md]: Should developers write tests for their changes?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+### 6. Run linting
+
+`$ npm run lint`
```
Comment (cat-haines) [inline CONTRIBUTING.md]: Why isn't this part of the tests?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+### src/scripts/config-page.js
+
+This is the main entry point for the code that runs on the generated config page,
+and is responsible for passing the injected config and other components to the
+`ClayConfig` class.
+
+
+## Building
+
+There are two ways to build Clay, production mode and development mode.
```
Comment (cat-haines) [inline CONTRIBUTING.md]: Might be good to explain the differences and why/when developers should use each mode

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+ - `$ npm run test` - Runs all tests once.
+ - `$ npm run test-debug` - Watches for file changes and re-runs all tests when it
+  detects a change. This also leaves an in instance of Google Chrome running.
+  If you click the "Debug" button in the top right of the page, a new tab will open.
+  You can then use the Google Chrome developer tools to debug your tests as they run.
+  Refreshing the page will re-run the tests.
+
+## Style Guide
+
+Have a look at the rest of the codebase for an example of how you should format your code.
```
Comment (cat-haines) [inline CONTRIBUTING.md]: I'm not sure I would describe "look at the rest of our codebase for an example" as being a style guide..

**Comment by cat-haines** [comment]
The language and information in this guide is really great, thanks for putting it together!! 

One addition that might be nice to see is where developers can go for support while developing for Clay (#clay on Slack, @pebbledev, forums, etc)

File: CONTRIBUTING.md (hunk 3 above)
Comment (keegan-lillo) [inline CONTRIBUTING.md]: They both run as part of CI but are broken up for local development so you can write tests without worrying about small linting issues like extra lines etc.

**Comment by keegan-lillo** [comment]
@cat-haines okay updated :)

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -189,3 +189,10 @@ following exceptions:
 
 The style guide is enforced using ESLint. Run `$ npm run lint` to ensure you are
 adhering to the style guide.
+
+## Getting Help
+
+ - Chat to us in the `#clay` channel on slack. http://slack.pbldev.io/
+ - Visit the [Pebble Forums](https://forums.pebble.com/)
+ - Email devsupport@getpebble.com
```
Comment (cat-haines) [inline CONTRIBUTING.md]: We're actually discouraging devsupport (with the exception of reporting actual bugs). Could you replace this with something like "Tweet at @pebbledev" (or something of that variety)

**Comment by keegan-lillo** [comment]
@cat-haines updated.

**Comment by cat-haines** [comment]
:+1: @keegan-lillo - One tiny comment on the "Getting Help" section, but this looks good.

You may also want to link to the contributing guide from the README.md to make it more discoverable.

**Pull request merged**